e2e/tests/negative/test_backup_listing.robot

```diff
@@ -0,0 +1,180 @@
+    ${workload_volume_name}=    get_workload_volume_name    ${workload_name}
+    ${backup_name}=    get_backup_name    ${backup_id}    ${workload_volume_name}
+    ${volume_name}=    generate_name_with_suffix    volume    ${volume_id}
+    check_restored_volume_checksum    ${volume_name}    ${backup_name}
+
+Check volume ${volume_id} data is intact
+    ${volume_name}=    generate_name_with_suffix    volume    ${volume_id}
+    check_data_checksum    ${volume_name}
+
+Check volume ${volume_id} data is data ${data_id}
+    ${volume_name}=    generate_name_with_suffix    volume    ${volume_id}
+    check_data_checksum    ${volume_name}    ${data_id}
+
+Get ${workload_kind} ${workload_id} volume data in file ${file_name}
+    ${workload_name}=    generate_name_with_suffix    ${workload_kind}    ${workload_id}
+    ${current_file_checksum}=    get_workload_pod_data_checksum    ${workload_name}    ${file_name}
+    [Return]  ${current_file_checksum}
+
+Volume ${volume_id} data should same as ${workload_kind} ${workload_id} volume
+    ${workload_name}=   generate_name_with_suffix    ${workload_kind}    ${workload_id}
+    ${workload_volume_name}=    get_workload_volume_name    ${workload_name}
+    ${backup_name}=    get_backup_name    ${random_backup_id}    ${workload_volume_name}
+    ${expected_file_checksum}=    get_restored_checksum    ${backup_name}
+    ${current_file_checksum}=  Get ${workload_kind} ${volume_id} volume data in file data
+    Should Be Equal  ${current_file_checksum}  ${expected_file_checksum}
+
+Perform backup 1001 times for deployment 0 volume
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        Create backup ${i} for deployment 0 volume
+        Delete snapshot ${i} of deployment 0 volume
+        IF    ${i} % 249 == 0
+            ${workload_name}=   generate_name_with_suffix    deployment    0
+            ${volume_name}=    get_workload_volume_name    ${workload_name}
+            purge_snapshot    ${volume_name}
+        END
+        Verify backup list contains backup no error for deployment 0 volume
+        Verify backup ${i} count for deployment 0 volume
+        Verify backup list contains backup ${i} of deployment 0 volume 
+    END
+
+Create pod ${pod_id} mount ${size} GB volume ${volume_id}
+    Create volume ${volume_id} with    size=${size}Gi    numberOfReplicas=3
+    Create persistentvolume for volume ${volume_id}
+    Create persistentvolumeclaim for volume ${volume_id}
+    Create pod ${pod_id} using volume ${volume_id}
+    Wait for pod ${pod_id} running
+
+Volume ${volume_id} backup ${backup_id} should be able to create
+    Create backup ${backup_id} for volume ${volume_id}
+    Verify backup list contains no error for volume ${volume_id}
+    Verify backup list contains backup ${backup_id} of volume ${volume_id}
+
+Write data to file in deployment 0
+    Write 100 MB data to file data in deployment 0
+
+Write 210 GB data in pod 2
```

Comment:
I don’t think it’s necessary to wrap it in another keyword. Using `Write 210 GB large data to file 0 in pod 2` directly in the test steps should be sufficient. WDYT?

Reply:
I agree with your suggestion, and I will make changes to this part.